albumentations/augmentations/functional.py

```diff
@@ -1424,3 +1425,67 @@ def spatter(
         raise ValueError("Unsupported spatter mode: " + str(mode))
 
     return img * 255
+
+@clipped
+@preserve_shape
+def dither(img: np.ndarray, nc: int) -> np.ndarray:
+    img = img.copy()
+    height = np.shape(img)[0]
+    is_rgb = True if len(np.shape(img)) == 3 else False
+
+    for y in range(height):
+        oldrow = img[y].copy()
+        quant_errors = []
+
+        if is_rgb:
+            # Turn into one list of length `width`, per channel (R, G, B)
+            #
+            # Use `tolist()` since operating on individual elements of an ndarray
+            # is very slow compared to a normal list.
+            channels = np.transpose(oldrow).tolist()
```

Comment:
Why we need to convert to list? Why do not use np.ndarray `channels = oldrow.transpose()`? Result would be the same, but faster.